CI: Remove Travis, fix TeamCity, rework E2E on CircleCI

gaetanmaisse opened the pull request:
## What I did

 - Remove TravisCI integration as it was no-op for 2 years
 - Fix TeamCity builds by using Node 12 and update Yarn commands (to Yarn 2 version)
 - Rework the E2E tests:
   - `e2e-tests-core` for "officially" supported frameworks : Vue + Angular (+ React + Web Components in e2e-tests-pnp)
   - `e2e-tests-extended` for other frameworks: Preact, Svelte, Riot, Rax etc
   - `e2e-tests-pnp` for some frameworks/setups compatible with PnP: React + WC
  - Upgrade to Cypress 6.8 and use Cypress docker image in CircleCI: Previously Cypress was installed manually to be sure to run with Node 10. We can now use the docker images provided by Cypress team (at least for the 6.8) because they are based on Node 12, the one we are using across the CI.
  - Merge multiple GitHub Actions related to DangerJS into one and fix the behaviour i.e. trigger a new run each time a label is set/unset on a PR.

 

## How to test

- All CircleCI + TeamCity builds should be 🟢

gaetanmaisse committed:
ci: update TeamCity settings to use Node 12 and handle Yarn 2

gaetanmaisse committed:
ci: remove Travis integration as it is a no-op since late 2019

gaetanmaisse committed:
ci: rework E2E tests in CircleCI

New layout:
 - `e2e-tests-core` for "officially" supported frameworks : Vue + Angular (+ React + Web Components in e2e-tests-pnp)
 - `e2e-tests-extended` for other frameworks: Preact, Svelte, Riot, Rax etc
 - `e2e-tests-pnp` for some frameworks/setups compatible with PnP: React + WC

gaetanmaisse committed:
fix(docs): add missing optional peerDep when using Docs with Web Components

gaetanmaisse commented:
Hey @Hypnosphi, I'm trying to fix the TeamCity builds but I don't know much about it, is there a simple way to run the builds with the updates I made in this branch? Because if I understand correctly it's always using the `next` version for the builds. 

I would be glad to discuss that on discord, feels free to ping me (same username as here ;))

Hypnosphi commented:
I changed the configuration to use settings from PR branch by default and re-started the build: https://storybook.beta.teamcity.com/buildConfiguration/Hosted_Root_TestWorkflow/126136

gaetanmaisse commented:
Thanks! I should be able to fix the failing jobs ;) and will let you know if I need some extra support.

gaetanmaisse committed:
chore: upgrade to Cypress 6.8 and use Cypress docker image in CircleCI

Previously Cypress was installed manually to be sure to run with Node 10.
We can now use the docker images provided by Cypress team (at least for the 6.8) because they are based on Node 12, the one we are using across the CI.

gaetanmaisse committed:
ci: use Node 12 in Netlify config

gaetanmaisse committed:
ci: merge multiple GitHub Actions related to DangerJS into one

Also, fix the behavior and trigger a new run each time a label is set/unset on a PR.
